controllers/common/common.go

```diff
@@ -43,29 +43,27 @@ const (
 	FileStorageTypeCache    FileStorageType = "cache"
 )
 
-func ResourceName(name string, componentKind v1alpha1.ComponentKind) string {
-	return name + "-" + string(componentKind)
-}
-
-func AdditionalResourceName(name, additionalName string, componentKind v1alpha1.ComponentKind) string {
-	if len(additionalName) == 0 {
-		return name + "-" + string(componentKind)
+// ResourceName returns the resource name for the given name and component kind.
+// If extraNames are provided, they will be appended to the resource name. For example,
+// - If the name is `my-cluster` and the component kind is `datanode`, the resource name will be `my-cluster-datanode`.
+// - If the name is `my-cluster` and the component kind is `datanode` and the extra names `read`, the resource name will be `my-cluster-datanode-read`.
+func ResourceName(name string, componentKind v1alpha1.RoleKind, extraNames ...string) string {
```

Comment:
My test result is `mycluster-read-datanode`, should it be `mycluster-datanode-read`?
```
package main

import (
	"fmt"
	"strings"
)

func main() {
	fmt.Println(ResourceName("mycluster", "datanode", "read"))
}

func ResourceName(name string, componentKind string, extraNames ...string) string {
	for _, extraName := range extraNames {
		if extraName != "" {
			name = name + "-" + extraName
		}
	}

	return strings.Join([]string{name, string(componentKind)}, "-")
}
```
output: 
```
go run main.go 
mycluster-read-datanode
```